warp10/src/main/java/io/warp10/script/NamedWarpScriptFunction.java

```diff
@@ -34,4 +34,8 @@ public void setName(String name) {
   public String getName() {
     return this.name;
   }
+
+  public final String refSnapshot() {
+    return "'" + this.name + "' " + WarpScriptLib.FUNCREF;
```

Comment:
The name needs to be encoded as SNAPSHOT does for STRINGs as it may contain any characters.